src/beamlime/applications/_nexus_helpers.py

```diff
@@ -1,327 +1,147 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
-import os
-from functools import partial
-from typing import (
-    Any,
-    Callable,
-    Literal,
-    Mapping,
-    NamedTuple,
-    Optional,
-    Tuple,
-    TypeVar,
-    Union,
-)
-
-Path = Union[str, bytes, os.PathLike]
-DTypeDesc = Literal["int", "float", "string"]
-
-
-class DatasetRecipe(NamedTuple):
-    name: str
-    dtype_description: DTypeDesc
-
-
-FlatBufferIdentifier = Literal["ev44", "f144", "tdct"]
-GROUP_RECIPES: Mapping[FlatBufferIdentifier, Tuple[DatasetRecipe, ...]] = {
-    "f144": (DatasetRecipe("timestamp", "int"), DatasetRecipe("value", "float")),
-    "tdct": (DatasetRecipe("name", "string"), DatasetRecipe("timestamps", "int")),
-    "ev44": (
-        DatasetRecipe("reference_time", "int"),  # In nanoseconds
-        DatasetRecipe("reference_time_index", "int"),
-        DatasetRecipe("time_of_flight", "int"),  # In nanoseconds
-        DatasetRecipe("pixel_id", "int"),
-    ),
-}
-
-
-NestedObjectT = TypeVar("NestedObjectT", dict, list[dict])
-
-
-def nested_shallow_copy(obj: NestedObjectT, *path_to_target) -> NestedObjectT:
-    """Shallow copy the nested dictionary and lists.
-
-    Examples
-    --------
-    >>> a = {'a': 1, 'b': {'b-a': 2}, 'c': {'c-a': 3}}
-    >>> b = nested_shallow_copy(a, 'b')
-    >>> b['b']['b-a'] = 3
-    >>> a['b']['b-a']  # The original dictionary is not changed
-    2
-    >>> b['b']['b-a']  # The copied dictionary is changed
-    3
-    >>> b['c']['c-a'] = 4
-    >>> a['c']['c-a']  # The original dictionary is also changed
-    4
-    """
-    from copy import copy
-
-    root_obj = obj.copy()
-    cur_parent = root_obj
-
-    try:
-        for idx in path_to_target:
-            cur_parent[idx] = copy(cur_parent[idx])
-            cur_parent = cur_parent[idx]
-    except (KeyError, TypeError) as e:
-        raise KeyError(f"Invalid path: {path_to_target}") from e
-
-    return root_obj
-
-
-def find_index(
-    obj: dict | list[dict],
-    *filter_or_indices: Callable | str | int,
-    _cur_idx: Optional[tuple[str | int, ...]] = None,
-) -> tuple[str | int, ...]:
```

Comment:
These helpers are also for finding attributes of the nexus not just children. But I guess we can put them back when they are actually needed...

Reply:
Yes we can add it back in that case. My feeling right now is that it is a too general helper function. It simplifies things to assume the kind of structure we expect (that is, nexus-json), instead of arbitrary nested lists/dictionaries.

Reply:
Yeah that's true. I made them generic assuming the structure is an arbitrary dictionary. 
Cause I thought... if we are so sure of the structure, we can just use scippnexus directly and no need to have these from the first place.

Reply:
Yeah I kind of have the same thought. Why can we not just use the stuff in `scippneutron` / `scippnexus` that deals with this nexus-json structure?

Reply:
It was because we didn't want to depend on the `assumption` of the how the nexus structure looks like,
and scippnexus is kind of depending on the `assumption` of how it looks like.

For example, we should be able to flexibly patch the structure to be compatible with the scippnexus if needed in the production I thought.... since it's meant to be run as a real-time application.